### ISSUE TITLE
IconColor has no effect when loading Icon from SVG in FluentNavLink

### ISSUE THREAD

**Issue by c0g1t8** [microsoft/fluentui-blazor#1240]
# 🐛 Bug Report

**IconColor** has no effect when loading **Icon** from SVG in **FluentNavLink**

## 💻 Repro or Code Sample

Create a project using the _fluentblazorwasm_ template. Build and run. Note the color of the icons are _Color.Accent_

## 🤔 Expected Behavior

Expect icon to have color specified.

## 🔦 Context

Discovered issue after adding _Microsoft.FluentUI.AspNetCore.Components.Icons_ to use different icons. _IconColor_ works as expected when specifying the _Icon_ by construction.  

```xml
<nav>
    <FluentNavMenu Width="250" Collapsible="true" Title="Custom navigation menu" Expanded="false">
        <FluentNavLink Href="/" Match="NavLinkMatch.All" Icon="@Icon.FromImageUrl("/images/home.svg")" IconColor="Color.Accent">Home</FluentNavLink>
        <FluentNavLink Href="mode" Icon="@(new Icons.Regular.Size20.Tv())" IconColor="Color.Accent">Counter</FluentNavLink>
        <FluentNavLink Href="counter" Icon="@Icon.FromImageUrl("/images/counter.svg")" IconColor="Color.Accent">Counter</FluentNavLink>
        <FluentNavLink Href="weather" Icon="@Icon.FromImageUrl("/images/weather.svg")" IconColor="Color.Accent">Weather</FluentNavLink>
    </FluentNavMenu>
</nav>```

## 🌍 Your Environment

* OS & Device: Windows 11
* Browser : Microsoft Edge, Google Chrome, Apple
* .NET:  .net8
*  FluentUi: v4.2.1

**Comment by vnbaaij** [comment]
Hi Gary,

IconColor will indeed not work when loading an SVG file. That is sort of 'by design' as we can't use the same way of processing the icon as we do with the 'construction' way.

In the other hand, the menu should not be using the FromImage way anymore. We decided to include the Icon package by default in the template sites but I seem to have forgotten the Wasm one. Will correct that for next release.

**Comment by c0g1t8** [comment]
I suspect the issue is related to loading the **img** element from an SVG file instead of rendering the **svg** element.

**Comment by c0g1t8** [comment]
> Hi Gary,
> 
> IconColor will indeed not work when loading an SVG file. That is sort of 'by design' as we can't use the same way of processing the icon as we do with the 'construction' way.
> 
> In the other hand, the menu should not be using the FromImage way anymore. We decided to include the Icon package by default in the template sites but I seem to have forgotten the Wasm one. Will correct that for next release.

Thanks for the quick response.  So, I suspect we will be getting updated templates too 😁

**Comment by c0g1t8** [comment]
@vnbaaij 

I like the idea of being able to load the icon from an svg file. It would provide a way for users to provide their own custom icons but still leverage the functionality built into **FluentIcon**. 

I'm thinking of extending FluentIcon by adding a FromUrl method that would render the svg element. Would you be willing to consider a PR, assuming I can get it all to work?

**Comment by vnbaaij** [comment]
Yes, of course! Just keep in mind that we can't 'interfere' with the rendering easily so things like color need to be specified in the source (url or image).

**Comment by vnbaaij** [comment]
Closing this as it works like this by design. Looking forward to the PR

**Comment by c0g1t8** [comment]
Comment made by @dvoituron for #1328 could be helpful for use in the PR.
https://github.com/microsoft/fluentui-blazor/issues/1328#issuecomment-1892245014

**Comment by vnbaaij** [comment]
Gary, can you elaborate. I don't understand what you mean with tyour last comment.

**Comment by c0g1t8** [comment]
> Gary, can you elaborate. I don't understand what you mean with tyour last comment.

I've been following the issues and saw a suggested solution for another problem that Denis suggested. I had not realized that the svg content could be specified in the base Icon constructor. It'sa much cleaner approach than what I was playing with. I was originally playing with a writing a new render fragment with the svg path.

Denis's approach would be more consistent with the existing implementation. So, I abandoned my work and intended to rewrite it. But, I got busy and dropped a note here as a reminder.
